lib/mcb.rb

```diff
@@ -113,13 +113,29 @@ def self.generate_apiv2_token(email:, encoding:, secret: nil)
   end
 
   def self.each_v1_course(opts)
+    endpoint = "/api/v1/#{get_recruitment_year(opts)}/courses"
+
     # This method can actually be entirely driven by the args provided, it is auto-configure!
-    iterate_v1_endpoint(**opts)
+    iterate_v1_endpoint(endpoint: endpoint, **opts)
   end
 
   def self.each_v1_provider(opts)
+    endpoint = "/api/v1/#{get_recruitment_year(opts)}/providers"
+
     # This method can actually be entirely driven by the args provided, it is auto-configure!
-    iterate_v1_endpoint(**opts)
+    iterate_v1_endpoint(endpoint: endpoint, **opts)
+  end
+
+  def self.get_recruitment_year(opts)
```

Comment:
If this method gets called before Rails is initialized it'll fail in potentially a non-obvious way. Can we make it more obvious to the user that Rails is required?